internal/webhook/kubernetes/warehouse/webhook_test.go

```diff
@@ -316,6 +317,69 @@ func TestValidateSpec(t *testing.T) {
 				)
 			},
 		},
+		{
+			name: "invalid when using constraint field",
```

Comment:
It's unnecessary to create dedicated test cases for testing individual fields and it adds quite a bit of clutter.

Amend the original invalid and valid test cases with additional invalid or valid subscriptions, respectively, and update assertions accordingly.

Reply:
This still needs to be fixed.